### ISSUE TITLE
print_literal lint warns when using fill/alignment directives

### ISSUE THREAD

**Issue by df5602** [rust-lang-nursery/rust-clippy#2634]
Hi,

Given something like this:
```
fn main() {
    println!("{: <width$}, World!", "Hello", width = 20);
}
```
clippy gives the following warning:
```
warning: printing a literal with an empty format string
 --> src/main.rs:2:37
  |
2 |     println!("{: <width$}, World!", "Hello", width = 20);
  |                                     ^^^^^^^
  |
  = note: #[warn(print_literal)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.192/index.html#print_literal
```

Is there a better way to achieve this without triggering the lint, or is it a false positive?

**Comment by phansch** [comment]
Looks like a false positive. Also these give a warning:

```rust
println!("Hello {:5}!", "x");
println!("Hello is {0:.5}", 0.01);
```

**Comment by phansch** [comment]
Going to take a look at this today

**Comment by phansch** [comment]
@mrecachinas Since you wrote the original `print_literal` lint: Do you know why the `print_literal` warns about these?

```rust
println!("10 / 4 is {}", 2.5);
println!("2 + 1 = {}", 3);
println!("2 + 1 = {:.4}", 3);
```
As far as I can see in the benchmarks, it's just about string literals, not integers or floats and `print_literal` should not warn about the three above. Is that correct?

**Comment by mrecachinas** [comment]
@phansch Good catch. I should have brought it up in the original issue/PR, but I (thought I) applied this to all literals with _empty_ format directives (e.g., `{}`). 

In that case, your first two examples correctly lint, but your last one doesn’t (`{:.4}` should yield no warning).

I’ll add some tests and take a look at it tomorrow.
